### ISSUE TITLE
"Text it to us!": An MMS gateway for gov't

### ISSUE THREAD

**Issue by daguar** [codeforamerica/project-ideas#20]
**BLUF**: A SaaS app that gives gov't agencies a phone # to accept multimedia text messages (pics, videos) and access them online; example usage: verification document submission
**Project Needs**: dev (I could do this), alpha govt partners (to understand business processes/user needs)
**Status**: idea/pie
**Details**:
##### The Problem

Gov't agencies require submission of lots of documents, often requiring mailing, faxing, or -- if we're lucky -- email. For many users of gov't services, particularly those of the safety net, submitting information this way is a hurdle.

Imagine someone who is homeless applying for SNAP: they need to submit evidence of an ID, but the sheer logistics of that are large, and unnecessarily so.
##### The Solution

Imagine if that client could use their smartphone (or that of a case worker) to take a picture of their hospital bracelet (valid ID for SNAP) and text it to a number which sends it to the Human Services Agency.

On the HSA end, staff would have a simple web app to access all MMS' that have been received, and archive those that they've reviewed.
##### Technical Feasibility

Twilio and a few other services offer an MMS API; this is pretty trivial.
##### Challenges

The proximate challenges are:

(1) Understanding the different gov't agency business processes where photo/MMS submission would be useful

(2) Understanding what (if any) privacy and security measures would need to be in place, and to what extent the underlying MMS infrastructure supports this. (That said, I think this could be punted on by simply getting informed consent to clients who wish to use the MMS service to submit info to the agency.)
##### Crazy-Ass Use Case: Video Consent

What if, instead of insane e-signature processes, a client could take a selfie video saying that they consent to something, and submit it to an agency. Imagine if this could have been feasible for Promptly's rollout.
##### Notes
- Not explicitly health-oriented, but definitely higher ROI in safety net programs

**Comment by lippytak** [comment]
Uhhh, this is a fantastic idea. One great user group in SF would be the [Homeless Outreach Team](http://sfhomeless.wikia.com/wiki/Homeless_Outreach_Team_-_HOT_Team_-_H.O.T._Team) (HOT). They literally wander the city and try to connect people with "the system." They focus on individuals who have no benefits, no caseworker, no doctor, etc.

One early challenge is that Twilio only [supports MMS via short codes](http://www.twilio.com/help/faq/sms/can-i-send-or-receive-mms-messages) right now and short codes [are at least $1k/mo](http://www.twilio.com/help/faq/short-codes/how-much-does-a-short-code-cost). If we decide to move this into prototype phase I could help find a partner in SF and ~$5k for a 3 month pilot. Another possible beneficiary is 311. Twilio.org may be amenable to giving us a short code for free on a temp basis (_nudge nudge_ @emilyville)

**Comment by daguar** [comment]
[Mogreet](https://developer.mogreet.com/) is another API option -- they use
shared short codes, which is (a) good because cheaper, (b) bad because they
require a keyword to route to your app, and so if someone mispelled "hsa"
their ID might go elsewhere.

On Fri, May 9, 2014 at 12:38 PM, Jake Solomon notifications@github.comwrote:

> Uhhh, this is a fantastic idea. One great user group in SF would be the Homeless
> Outreach Teamhttp://sfhomeless.wikia.com/wiki/Homeless_Outreach_Team_-_HOT_Team_-_H.O.T._Team(HOT). They literally wander the city and try to connect people with "the
> system." They focus on individuals who have no benefits, no caseworker, no
> doctor, etc.
> 
> One early challenge is that Twilio only supports MMS via short codeshttp://www.twilio.com/help/faq/sms/can-i-send-or-receive-mms-messagesright now and short codes are
> at least $1k/mohttp://www.twilio.com/help/faq/short-codes/how-much-does-a-short-code-cost.
> If we decide to move this into prototype phase I could help find a partner
> in SF and ~$5k for a 3 month pilot. Another possible beneficiary is 311.
> Twilio.org may be amenable to giving us a short code for free on a temp
> basis (_nudge nudge_ @emilyville https://github.com/emilyville)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/codeforamerica/health-project-ideas/issues/20#issuecomment-42705329
> .

## 

Dave Guarino
Consultant, Health Vertical
(2013 Fellowship alumnus)
Code for America http://www.codeforamerica.org/
dave@codeforamerica.org
LinkedIn http://www.linkedin.com/in/daveguarino/ |
GitHubhttps://github.com/daguar/

**Comment by junosuarez** [comment]
Most carriers host their own SMS- and MMS-to-email gateways, meaning eg that you could send an MMS to an email address, eg http://www.att.com/esupport/article.jsp?sid=KB63037&cv=820#fbid=sxvaOD9SUwV

Hypothesis to validate: MMS is better than plain email. Possible pros: may be cheaper to send, may support broader range of devices/carriers, may be easier to send from phone's UI. Possible cons: more difficult to process, more expensive to process.

**Comment by daguar** [comment]
Word! Good to know.

For our use case, we'd want users to be able to MMS **to** the HSA. I assume the email gateway doesn't help us there, right?

Or could HSA have a phone number with an email gateway where they can access the MMS sent to them in a browser via the email address?

**Comment by lippytak** [comment]
A long discussion of a specific potential use case: https://github.com/codeforamerica/calfresh-and-so-clean/issues/21

**Comment by daguar** [comment]
Boom — game-changer. Twilio now allows MMS on all phone numbers (not just short codes):

https://www.twilio.com/blog/2014/09/introducing-twilio-mms-nt.html

**Comment by lippytak** [comment]
Francesca of the SF Food Bank pitched me this idea in the context of https://github.com/codeforamerica/calfresh-and-so-clean. Apparently LOTS of partner orgs have clients going through and could feasibly submit a ~5-10 min initial CalFresh app but then the big bottleneck is verification docs. Obviously they won't have everything on the spot. But Francesca is pretty confident that clear instructions to "go back home, get these three docs, and text pictures of them to the county" would work. @daguar @alanjosephwilliams we should make this happen...let's chat about it Mon and if we decide to do it now I'll figure out internal business process stuff (what EXACTLY do you need to find the right case?).

**Comment by cescacosta** [comment]
YES! I already tell clients to text me pics of their documents so that I can email them to the county. People think it's a lot easier than fax or snail mail (since there's no need to make copies first) and even my clients who don't know how to use email seem to be comfortable with sending picture texts. 

The challenge as I see it is; how can HSA identify who the document came from, especially before the person has case number? And of course, there's always the issue of security.  

This is a crazy idea, but--could Clean ever have the capacity to auto text someone their case # once they have been assigned it (if they consent to text messages)? No idea what that would look like on your end, but if they could get their case number within a day or two (or 5?), they could send a text with the missing docs and the case number. Otherwise we have the issue of texting SSNs into the universe, which as @lippytak and I discussed, is probably not a good idea, or we have to find some other method of attaching the person's documents to their case. I just asked my HSA contact and he says they can't search by phone number, but they could maybe do it by address or last 4 of SSN. The danger, he says, is that if they can't ID the docs right away, they might get pended indefinitely...

**Comment by lippytak** [comment]
hey hey! Welcome to Github @cescacosta! This is great...definitely justifies the need. Seems like the open questions are:
- What's the minimal amount of info to identify a case ~95% of the time or more. Last name and last 4 of SSN could work.
- Where does the message go and how does it make it to the EW? Can we use the existing email pathway? @cescacosta in your experience how does it work when you submit verification docs via email? How do you know if/when they're incorporated into the case?
- What happens when we can't identify the case? How can we alert clients that HSA didn't get the appropriate info?

Great point about getting case #s to clients. If clients had easy access to their case # it would make a lot of this moot and would be useful for many other purposes. I'll simmer on this but it will be tricky...

**Comment by cescacosta** [comment]
Maybe we can ask Carissa about identifying a case with last name and last 4 of SSN? I know that at least in the case of our Spanish and Chinese speaking clients, some last names are so common that it might make sense to at least include first, last and last 4 of SSN. 

I think the important thing here is to make it easy enough to EW to follow through...it sucks to have to put it that way, but if it's going to take a lot of digging for them to ID the verif, they might just make say it wasn't clear who it belonged to and move on to the next thing. This is what I sorta gathered from my conversation with Andrew yesterday (not that he said that explicitly, but you know...). 

In my experience, emailing documents (with a case number or SSN and DOB) has worked pretty well. Everything that goes to that inbox is tasked out to individual workers. I haven't had many issues with things not making it through (that I know of) but that said, I get close to zero responses from them via email. When we first got that email address for "follow-up" I tried emailing case issues to that email address but got no response. Every once in a great while I'll get a response, but that's only if the photo was too blurry to use, or they couldn't open the file for whatever reason. So typically, no new is good news out of that inbox. I don't know if every verification that couldn't be seen on their end made it through though, just because we don't follow up with every single one of our clients after sending docs. Having a date on the email (and an e-trail) has been immensely helpful because if HSA ever says they didn't get something, I have an email with a date on it that proves that it was sent to them (of course that relies on the client calling us to say that county says that they didn't get what we sent). That has come in handy more than once. Overall, though, I suspect 95ish% success rate. The verifications take 5-7 days to be added to their case but they're considered "in" from the date they are sent. As in, if a client has 10 days to submit the verification and I send it on the 9th day, we won't get confirmation that it has been received until 5-7 days later when it is processed, but ultimately, it got there on time to approve the case. 

Ok I gotta run to office hours now, but so excited to get this conversation going! I think clean+easy doc submission could be a game changer for places that are already doing a simple intake process with clients (like WIC offices, health clinics, school social workers...etc.). Let's do it!

**Comment by daguar** [comment]
This is so, so great and I have so much to say and omg but I'm posting here to say that I'm speaking with the product manager for MMS at Twilio tomorrow so suss out some of the basic questions. Hopefully that'll get us to a place where we feel good to sprint to an alpha.

**Comment by cescacosta** [comment]
YES!!! I love it!

**Comment by jtashea** [comment]
I've been playing around with a similar idea in Maryland. Currently, the Department of Juvenile Services (ie. the youth criminal justice system) does not allow their case workers or parole officers cell phones, which means that staff can't text their clients (ages 10-21) about court dates, other important meetings, or anything. Why is this important? In Baltimore City, a 2010 report said that the plurality of youth were getting sent to out-of-home placement (ie. prison, group home, or treatment center) not because of the original offense but because of a technical violation (like a failure to appear in court) or a parole violation. What this means is that we are incarcerating youth in Maryland not because of their delinquent act, but because of their failure to navigate a complex criminal justice system that can't text. (What 10-21 year old talks on the phone anyway?)

I believe that if a simple texting API (SMS or MMS) were created for case workers, they could better keep up with their kids getting them to court more often, which ultimately means fewer kids in prison that don't need to/shouldn't be there. I know that CfA did something similar in Atlanta with adults, and it's replication in this context would be extremely valuable to both government and these youth. The app (if MMS) could also work for probation officers making sure the GPS tracking on the kid's ankle is accurate (lots of technical problems with those). If there was any question where the kid was, he could just snap a photo of the street sign on the corner and confirm location, again pushing down confusion and violations while at the same time allowing the youth to be more proactive in his probation.

Studies show that people's feelings about whether or not they were treated justly by the criminal justice system has less to do with outcome (ie. sentencing) and more to do with feeling like you were treated fairly. An app like this would be a huge step in treating these kids better as they interact with the criminal justice system.

**Comment by lippytak** [comment]
@jtashea this is a fantastic and interesting perspective that's really rare on Github, so HUGE THANKS for hopping in!

I wonder if we could validate the problems/solutions using existing tools rather than building new things form scratch. Here are a few that come to mind:
- You can send/receive text messages through Google Voice: https://support.google.com/voice/answer/115116?hl=en. I remember back in the day you could set this up with GChat so it felt like you were chatting on your computer when you were really sending and receiving text messages.
- GroupMe (https://groupme.com/) is targeted more at group conversations (maybe a case worker would want to message all of their clients at once?), but basically it gives you a text box in your web browser that you can use to send/receive messages from groups of people.
- Our team has been using [Textit.in](http://www.textit.in) to do text message surveys and outbound text message blasts. There isn't really an elegant chat-like interface but it gets the job done of sending and receiving text messages via a web browser.
- And last, most phone carriers support text-to-email out of the box, so you can send an MMS to an email address. This could support the MMS use case you highlighted below (which is FASCINATING and so important...). Try texting a picture to your email address right now and I'll bet it will work.

So these are definitely all hacky solutions and certainly not developed with these use cases in mind...but maybe you could use these to help shed light on the problem and best possible solutions. For example, I wonder if providing a good web interface would just postpone solving the underlying issue that case workers don't have access to the right tools to talk to their clients.

@jtashea do you work directly with any juvenile case workers? Definitely keep us posted if you're able to move this forward. Also, my cell is (510) 206-8727 - feel free to test out any of the services above using me as a guinea pig.

**Comment by daguar** [comment]
MVP spec is being developed by @lippytak here https://github.com/codeforamerica/health/issues/34

**Comment by jtashea** [comment]
@lippytak This is great feedback. I do a lot of work with the Maryland Dept. of Juvenile Services, and this is an idea I've been kicking around with them for awhile. They very much want to implement an inexpensive tool that handles texting. I think each of these products offers a piece of the puzzle. Below are a couple of quick thoughts on these ideas and where there may be hurdles. Please feel free to correct any misreads on my part.

Google Voice: The number of texts is capped; this would create a workability issue. If you want to get case workers to use a new tool, it's just gotta work. If each has seven kids at a time, and they can't easily communicate over this medium with them because they ran out of texts, they just aren't going to use the technology. Also, it would seem onerous to me to have them count how many texts they've sent in a month. Is there a way around the cap?

GroupMe: The application might not meet the needs of a caseworker or the law. Due to stringent privacy rules protecting youth in the juvenile system, a group text would make public that youth's identity. I didn't see much in the way of anonymous options. So, that would be a legal and practical hurdle to navigate.

TextIt: I like this model. One initial concern would be the ability to navigate unique problems that come up during a youth's case. Certainly, TextIt would be perfect for court date and counseling reminders. "Tomorrow is your probation hearing. Will you be there? Type 'yes' or 'no'" and so on. Perhaps that's the place to start with this project, automated court reminders alone would be a massive improvement. 

All: It is unclear to me if any of these programs create a thorough paper trail. For the sake of court and DJS follow up, both entities would want to have a clear, accessible, chronological case-by-case paper trail to know if case workers are doing their job and so that the right party is held accountable when there is a lapse (either by agency or by youth). Second, what type of security standards are these programs using? As I mentioned above, a youth's privacy is paramount while in the system. I wouldn't want to start down a path that improves connectivity and jeopardize a youth's identity in the process.

These are my initial thoughts. Please let me know what you think @lippytak. I think it makes sense to use something already out there, I just want to make sure that we entertain these unique hurdles around the jj system. Thanks again.

**Comment by lippytak** [comment]
@jtashea you're right on about these considerations...just have a couple more small thoughts.

For GroupMe I was thinking you could have single person groups if case workers just want a way to have an SMS conversation with youth via a computer.

For Textit.in you can go to Messages > Inbox to see a full view of any message that aren't captured by the existing flows (surveys), so there _could_ be an internal business process to monitor those.

Re. the paper trail, none of these are going to have an easy out-of-the-box way to support your specific use case but all of them certainly give you access to the underlying data if you need it. But this would take additional work to figure out and probably a developer to automate this process if it's needed frequently and by many different people.

Re. privacy, the unfortunate fact is that SMS isn't encrypted so you really can't consider it secure. This is equally true of any of these services. I don't know anything about the info security and privacy laws in this domain, but in general it's certainly best practice to get explicit consent from the individual before any SMS conversation. Okay...think that's it for now. Good luck!

**Comment by jtashea** [comment]
@lippytak I appreciate the thoughtful feedback. I think you are right on the privacy stuff, at this point it's a wash on SMS. None of these ideas raise new privacy concerns that don't already exist in the current system.

Based on your other points, the only one that stands out as an unresolved hurdle would be the paper trail. To get a DJS like entity on board, there would need to be an easy and accessible way to pull up the history. The court, prosecution, and defenders would expect this as well. From a developer's perspective, how involved would this be to put together in its simplest form? Thanks!

**Comment by lippytak** [comment]
I'd say "pretty easy" though am now obliged to say I'm not a developer! But it would be easy to see all messages exchanged between two people and the timestamps...or something similar. Feel free to email me jacob@codeforamerica.org if you want to go into more detail about this work.
